test/quantization/test_workflow_module.py

```diff
@@ -670,3 +677,102 @@ def test_fq_serializable(self):
         loaded_dict = torch.load(b)
         for key in state_dict:
             self.assertEqual(state_dict[key], loaded_dict[key])
+
+def _get_buffer_ids(module):
+    """
+    Object addresses stay constant if and only if all modifications are in-place
+    """
+    return [id(v) for k, v in module._buffers.items()]
+
+class TestDistributed(QuantizationTestCase):
+
+    def test_observers_preserve_buffers(self):
+        """
+        Tests that observers only modify buffers in place. Note: this is important
+        because nn.DataParallel depends on this assumption to work correctly.
+        However, DataParallel does not expose IDs of the replicas, so we test it
+        without DataParallel in order to easily access the object IDs.
```

Comment:
Can we test with dataparallel?

Reply:
The new test suite already has three tests:
* two unit tests which test the actual issue.  It's not possible to test this specific issue in DataParallel because the buffers of the replicas are hidden.
* an integration test, using DataParallel.